kyuubi-server/src/main/scala/org/apache/kyuubi/server/api/v1/BatchesResource.scala

```diff
@@ -203,6 +204,20 @@ private[v1] class BatchesResource extends ApiRequestContext with Logging {
       KyuubiApplicationManager.uploadWorkDir,
       resourceFileMetadata.getFileName)
     batchRequest.setResource(tempFile.getPath)
+
+    val subresoucesMap = batchRequest.getSubresourcesMap
+    val transformedSubMap = subresoucesMap.entrySet().asScala.map { a =>
+      val targetConfKey = a.getKey
+      val subresourceFileName = a.getValue
+      val filePart = multiPart.getField(subresourceFileName)
+      val subresourceFile = Utils.writeToTempFile(
+        filePart.getValueAs(classOf[InputStream]),
+        KyuubiApplicationManager.uploadWorkDir,
+        filePart.getContentDisposition.getFileName)
+      (targetConfKey, subresourceFile.getPath)
+    }.toMap.asJava
+    batchRequest.setTransformedSubresourcesMap(transformedSubMap)
```

Comment:
I'm confused here, for the main resource, we just replace the reference directly, for extra resources, why not find the references and replace them directly too?

Reply:
Yes, changed to replacing them directly.